src/test/java/htsjdk/samtools/SAMFileWriterFactoryTest.java

```diff
@@ -304,4 +320,75 @@ public void testAsync() throws IOException {
         writer = builder.setUseAsyncIo(false).makeWriter(header, false, outputFile, referenceFile);
         Assert.assertFalse(writer instanceof AsyncSAMFileWriter, "testAsync option=unset");
     }
+
+    @Test
+    public void testMakeWriterForSamExtension() throws IOException {
+        final File tmpFile = File.createTempFile("testMakeWriterForSamExtension", ".sam");
+        tmpFile.deleteOnExit();
+        final SAMFileWriter samFileWriter = new SAMFileWriterFactory().makeWriter(new SAMFileHeader(), true, tmpFile, null);
+        samFileWriter.close();
+
+        final FileInputStream fis = new FileInputStream(tmpFile);
+        for (byte b:"@HD\tVN:".getBytes()) {
+            Assert.assertEquals((byte)(fis.read() & 0xFF), b);
+        }
+        fis.close();
+    }
+
+    @Test
+    public void testMakeWriterForBamExtension() throws IOException {
+        final File tmpFile = File.createTempFile("testMakeWriterForBamExtension", ".bam");
```

Comment:
Fixed. 
One annoying moment: BamFileIoUtils.BAM_FILE_EXTENSION includes dot character, whereas SamReader.Type file extensions don't. This is confusing but this is probably for another PR.